server/routes/selectGender/selectGenderPage.ts

```diff
@@ -0,0 +1,104 @@
+/* eslint-disable max-classes-per-file */
+import { Request, Response } from 'express'
+
+import PlaceOnReportService from '../../services/placeOnReportService'
+import adjudicationUrls from '../../utils/urlGenerator'
+import { FormError } from '../../@types/template'
+import { User } from '../../data/hmppsAuthClient'
+
+type PageData = {
+  error?: FormError
+  genderSelected?: string
+  draftId?: number | null
+  readApiGender?: string
+}
+
+export enum PageRequestType {
+  ORIGINAL,
+  EDIT,
+}
+
+class PageOptions {
+  constructor(private readonly pageType: PageRequestType) {}
+
+  isEdit(): boolean {
+    return this.pageType === PageRequestType.EDIT
+  }
+}
+
+export default class SelectGenderPage {
+  pageOptions: PageOptions
+
+  constructor(pageType: PageRequestType, private readonly placeOnReportService: PlaceOnReportService) {
+    this.pageOptions = new PageOptions(pageType)
+  }
+
+  private renderView = async (res: Response, pageData: PageData): Promise<void> => {
+    const { error, draftId, readApiGender } = pageData
+
+    res.render(`pages/selectGender`, {
+      errors: error ? [error] : [],
+      genderSelected: readApiGender,
+      cancelButtonHref: getCancelHref(this.pageOptions, draftId),
+      hintText:
+        'This is the gender the prisoner identifies as. We’re asking this because there’s no gender specified on this prisoner’s profile.',
+    })
+  }
+
+  view = async (req: Request, res: Response): Promise<void> => {
+    const draftId = Number(req.params.draftId) || null
+    const { user } = res.locals
+
+    let readApiGender: string = null
+    if (this.pageOptions.isEdit()) {
+      readApiGender = await this.getPreviouslySelectedGenderFromApi(draftId, user)
```

Comment:
re edit gender, i think its only available if the prisoner was undefined previously?